pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -861,16 +861,33 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
         return index < _bptIndex ? index : index.sub(1);
     }
 
-    function _dropBptItem(uint256[] memory amounts)
+    /**
+     * @dev Remove the item at `_bptIndex` from an arbitrary array (e.g., amountsIn).
+     */
+    function _dropBptItem(uint256[] memory amounts) internal view returns (uint256[] memory) {
+        uint256[] memory amountsWithoutBpt = new uint256[](amounts.length - 1);
+        for (uint256 i = 0; i < amountsWithoutBpt.length; i++) {
+            amountsWithoutBpt[i] = amounts[i < _bptIndex ? i : i + 1];
+        }
+
+        return amountsWithoutBpt;
+    }
+
+    /**
+     * @dev Remove the item at `_bptIndex` from an array assumed to represent the current balances.
+     * Since the number at that index is the current BPT balance, use it to compute and return
+     * the virtualSupply.
+     */
+    function _dropBptItemFromBalances(uint256[] memory balances)
         internal
         view
-        returns (uint256 virtualSupply, uint256[] memory amountsWithoutBpt)
+        returns (uint256 virtualSupply, uint256[] memory balancesWithoutBpt)
     {
-        virtualSupply = _getVirtualSupply(amounts[_bptIndex]);
+        virtualSupply = _getVirtualSupply(balances[_bptIndex]);
 
-        amountsWithoutBpt = new uint256[](amounts.length - 1);
-        for (uint256 i = 0; i < amountsWithoutBpt.length; i++) {
-            amountsWithoutBpt[i] = amounts[i < _bptIndex ? i : i + 1];
+        balancesWithoutBpt = new uint256[](balances.length - 1);
+        for (uint256 i = 0; i < balancesWithoutBpt.length; i++) {
+            balancesWithoutBpt[i] = balances[i < _bptIndex ? i : i + 1];
```

Comment:
```suggestion
        balancesWithoutBpt = _dropBptItem(balances);
```

Reply:
I think I am not getting enough sleep :)